tools/query/src/query_view.rs

```diff
@@ -12,16 +12,26 @@ pub async fn get_view(
     client.view_ext(function_id, type_args, args).await
 }
 
-// helper to turn a serde_json value to string
-// TODO: must be a better way
-pub fn display_view(res: Vec<Value>) -> Result<String> {
-    let values_to_string = res.iter().map(|v| v.to_string()).collect::<Vec<_>>();
-    if values_to_string.len() > 1 {
-        Ok(format!("[{}]", values_to_string.join(", ")))
-    } else {
-        Ok(format!(
-            "[{}]",
-            values_to_string.first().expect("api didn't return a value")
-        ))
+pub async fn fetch_and_display(
```

Comment:
I would keep display_view() as a separate function.
If you want this to be a one step then I would call  out to that function.